src/security/credentials.rs

```diff
@@ -209,29 +208,28 @@ impl Drop for ServerCredentialsBuilder {
     }
 }
 
-/// Server-side SSL credentials.
-///
-/// Use [`ServerCredentialsBuilder`] to build a [`ServerCredentials`].
-pub struct ServerCredentials {
-    creds: *mut grpc_server_credentials,
-}
-
-unsafe impl Send for ServerCredentials {}
-
 impl ServerCredentials {
-    pub(crate) unsafe fn frow_raw(creds: *mut grpc_server_credentials) -> ServerCredentials {
-        ServerCredentials { creds }
-    }
-
-    pub fn as_mut_ptr(&mut self) -> *mut grpc_server_credentials {
-        self.creds
-    }
-}
-
-impl Drop for ServerCredentials {
-    fn drop(&mut self) {
+    /// Creates the credentials using a certificate config fetcher. Use this
+    /// method to reload the certificates and keys of the SSL server without
+    /// interrupting the operation of the server. Initial certificate config will be
+    /// fetched during server initialization.
+    pub fn with_fetcher(
+        fetcher: Box<dyn ServerCredentialsFetcher + Send + Sync>,
+        cer_request_type: CertificateRequestType,
+    ) -> Self {
+        let fetcher_wrap = Box::new(fetcher);
+        let fetcher_wrap_ptr = Box::into_raw(fetcher_wrap);
         unsafe {
-            grpc_sys::grpc_server_credentials_release(self.creds);
+            let opt = grpcio_sys::grpc_ssl_server_credentials_create_options_using_config_fetcher(
+                cer_request_type.to_native(),
+                Some(server_cert_fetcher_wrapper),
+                fetcher_wrap_ptr as _,
+            );
+            let mut creds = ServerCredentials::from_raw(
+                grpcio_sys::grpc_ssl_server_credentials_create_with_options(opt),
+            );
+            creds._fetcher = Some(Box::from_raw(fetcher_wrap_ptr));
```

Comment:
`_fetcher` isn't fully gated with `_secure`?

Reply:
No, grpc c core will not delete it.

Reply:
Oh, you mean whether it can be accessed in "credentials.rs". Yes, it can. The file is also feature gated by "_secure".